solvebio/query.py

```diff
@@ -880,26 +904,38 @@ def join(self, query_b, key, key_b=None, prefix="b_", always_prefix=True):
             else:
                 name = field.name
 
+            query_b_prefixed_fields.append(name)
+
             if name in existing_field_names:
-                logger.warning("Field '{}' will be overwritten in the join results")
+                logger.warning("Field '{}' will be overwritten in the join results".format(name))
 
             target_fields.append({
                 "name": name,
                 "title": field.title,
                 "data_type": field.data_type,
                 "ordering": field.ordering,
-                "is_list": False,
+                "is_list": True,
                 "is_transient": False,
                 "expression": """
-                    get(record, "{}.{}")
+                    [get(item, "{}.{}") for item in record.exploded_join_field]
```

Comment:
I'm not sure if the double explode is needed. Can't we just create a list of values using `[get(item, "...") for item in record. query_b_join_field_name`? and then explode in post_annotate

Reply:
You are right. Done.